gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -914,6 +933,99 @@ public Operation delete(final AddressId addressId, OperationOption... options) {
+            @Override
+            public com.google.api.services.compute.model.Snapshot call() {
+              return computeRpc.getSnapshot(snapshot, optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Snapshot.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Page<Snapshot> listSnapshots(SnapshotListOption... options) {
+    return listSnapshots(options(), optionMap(options));
+  }
+
+  private static Page<Snapshot> listSnapshots(final ComputeOptions serviceOptions,
+      final Map<ComputeRpc.Option, ?> optionsMap) {
+    try {
+      ComputeRpc.Tuple<String, Iterable<com.google.api.services.compute.model.Snapshot>> result =
+          runWithRetries(new Callable<ComputeRpc.Tuple<String,
+              Iterable<com.google.api.services.compute.model.Snapshot>>>() {
+            @Override
+            public ComputeRpc.Tuple<String,
+                Iterable<com.google.api.services.compute.model.Snapshot>> call() {
+              return serviceOptions.rpc().listSnapshots(optionsMap);
+            }
+          }, serviceOptions.retryParams(), EXCEPTION_HANDLER);
+      String cursor = result.x();
+      Iterable<Snapshot> snapshots = Iterables.transform(
+          result.y() == null ? ImmutableList.<com.google.api.services.compute.model.Snapshot>of()
+              : result.y(),
+          new Function<com.google.api.services.compute.model.Snapshot, Snapshot>() {
+            @Override
+            public Snapshot apply(com.google.api.services.compute.model.Snapshot snapshot) {
+              return Snapshot.fromPb(serviceOptions.service(), snapshot);
+            }
+          });
+      return new PageImpl<>(new SnapshotPageFetcher(serviceOptions, cursor, optionsMap), cursor,
+          snapshots);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Operation deleteSnapshot(final SnapshotId snapshot, OperationOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Operation answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Operation>() {
+            @Override
+            public com.google.api.services.compute.model.Operation call() {
+              return computeRpc.deleteSnapshot(snapshot.snapshot(), optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Operation.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Operation deleteSnapshot(final String snapshot, OperationOption... options) {
+    return deleteSnapshot(SnapshotId.of(snapshot));
```

Comment:
Probably not worth changing, but it seems like you could have moved the runWithRetries logic here and called this deleteSnapshot method from the other supplying snapshot.snapshot() as a parameter.  That avoids creating a snapshot ID when it isn't necessary.